Objects/longobject.c

```diff
@@ -262,6 +262,35 @@ _PyLong_FromSTwoDigits(stwodigits x)
     return _PyLong_FromLarge(x);
 }
 
+int
+_PyLong_AssignValue(PyObject **target, Py_ssize_t value)
+{
+    PyObject *old = *target;
+    if (IS_SMALL_INT(value)) {
+        *target = get_small_int(Py_SAFE_DOWNCAST(value, Py_ssize_t, sdigit));
+        Py_XDECREF(old);
+        return 0;
+    }
+    else if (old != NULL && PyLong_CheckExact(old) &&
```

Comment:
This only works for positive integers.
Given that it is only used for `range` iteration, that's fine for now.
But could you add a comment, so it doesn't get overlooked when we implement https://github.com/faster-cpython/ideas/discussions/147